### PR TITLE
fix(staking): use reserve for bumping fee

### PR DESCRIPTION
**Pull request by dev-pvl**
## Description

Fixed an issue where the gas limit was incorrect during the fee bumping.  

## Related Issue

Resolve 
https://github.com/trezor/trezor-suite/issues/12183

## Screenshots:
Stake

unstake

### CODE REVIEW

File: packages/suite/src/utils/suite/stake.ts [hunk 1]
```diff
@@ -563,3 +564,30 @@ export const getDaysToAddToPoolInitial = (validatorsQueue?: ValidatorsQueue) =>
 
     return daysToWait <= 0 ? 1 : daysToWait;
 };
+
+export const stakeTypes: StakeType[] = ['stake', 'unstake', 'claim'] as const;
+
+export const isStakeType = (value: any): value is StakeType => {
+    return stakeTypes.includes(value);
+};
```
Comment (tomasklim) [inline packages/suite/src/utils/suite/stake.ts]: Use already existing `isStakeTypeTx` method

File: packages/suite/src/reducers/suite/modalReducer.ts [hunk 2]
```diff
@@ -97,4 +99,16 @@ const modalReducer = (state: State = initialState, action: Action): State => {
     }
 };
 
+export const selectTxStakeName = (state: AppState) => {
```
Comment (tomasklim) [inline packages/suite/src/reducers/suite/modalReducer.ts]: I think it does not belong to this file

File: packages/suite/src/reducers/suite/modalReducer.ts [hunk 3]
```diff
@@ -97,4 +99,16 @@ const modalReducer = (state: State = initialState, action: Action): State => {
...
+    if (
+        state.modal.context === MODAL.CONTEXT_USER &&
+        state.modal.payload.type == 'transaction-detail'
+    ) {
+        const txName = state.modal.payload.tx.ethereumSpecific?.parsedData?.name?.toLowerCase();
```
Comment (tomasklim) [inline packages/suite/src/reducers/suite/modalReducer.ts]: can it be simplified somehow?

File: packages/suite/src/utils/suite/stake.ts [hunk 4]
```diff
@@ -563,3 +564,30 @@ export const getDaysToAddToPoolInitial = (validatorsQueue?: ValidatorsQueue) =>
...
+
+export const getUnstakeAmount = (ethereumData: string) => {
```
Comment (tomasklim) [inline packages/suite/src/utils/suite/stake.ts:565]: add tests

File: packages/suite/src/utils/suite/stake.ts [hunk 5]
```diff
@@ -563,3 +564,30 @@ export const getDaysToAddToPoolInitial = (validatorsQueue?: ValidatorsQueue) =>
...
+    const dataBuffer = Buffer.from(ethereumData, 'hex');
+
+    return hexToNumberString('0x' + dataBuffer.subarray(4, 36).toString('hex'));
+};
+
+export const handleStakeTransaction = (values: FormState, txStakeName: StakeType) => {
```
Comment (tomasklim) [inline packages/suite/src/utils/suite/stake.ts:571]: test

File: packages/suite/src/hooks/wallet/form/useCompose.ts [hunk 6]
```diff
@@ -37,6 +41,7 @@ export const useCompose = <TFieldValues extends FormState>({
     getValues,
     formState: { errors },
     clearErrors,
+    txStakeName,
```
Comment (tomasklim) [inline packages/suite/src/hooks/wallet/form/useCompose.ts:44]: I am not fan of this, I will take a look how to improve it.

**Review by tomasklim — CHANGES_REQUESTED** [review]
It does not work for claim
It does not fix review modal

File: packages/suite/src/hooks/wallet/form/useCompose.ts (hunk 6 above)
Comment (dev-pvl) [reply packages/suite/src/hooks/wallet/form/useCompose.ts:44]: Me neither to be honest. Just didn't find a way how to do it in a better way. Waiting for your suggestions.

File: packages/suite/src/utils/suite/stake.ts (hunk 5 above)
Comment (dev-pvl) [reply packages/suite/src/utils/suite/stake.ts:571]: https://github.com/trezor/trezor-suite/pull/12538/commits/d874f631c0ab6ad1cd52a7228cd1da9de192a84d

File: packages/suite/src/utils/suite/stake.ts (hunk 4 above)
Comment (dev-pvl) [reply packages/suite/src/utils/suite/stake.ts:565]: https://github.com/trezor/trezor-suite/pull/12538/commits/d874f631c0ab6ad1cd52a7228cd1da9de192a84d

File: packages/suite/src/reducers/suite/modalReducer.ts (hunk 3 above)
Comment (dev-pvl) [reply packages/suite/src/reducers/suite/modalReducer.ts]: refactored https://github.com/trezor/trezor-suite/pull/12538/commits/d874f631c0ab6ad1cd52a7228cd1da9de192a84d

File: packages/suite/src/reducers/suite/modalReducer.ts (hunk 2 above)
Comment (dev-pvl) [reply packages/suite/src/reducers/suite/modalReducer.ts]: Could you help me to find a proper place for this function? The project is pretty large, so it's hard to see it yet.

File: packages/suite/src/utils/suite/stake.ts (hunk 1 above)
Comment (dev-pvl) [reply packages/suite/src/utils/suite/stake.ts]: https://github.com/trezor/trezor-suite/pull/12538/commits/d874f631c0ab6ad1cd52a7228cd1da9de192a84d

**Comment by dev-pvl** [comment]
It should work for claim now. 

claim

**Comment by tomasklim** [comment]
It increases the gas limit correctly 👍 

@dev-pvl 

However:
It modifies the amount of unstake tx, it should just show the amount but not broadcast it.
https://holesky2.trezor.io/tx/0xdcb971d90c8d5fe25e609a94f1ad87b1e94ebe71da47bd038c0a6d41e639935e

ok amount 
Screenshot 2024-06-25 at 19 24 12

but it fails because the amount is actually send
Screenshot 2024-06-25 at 19 32 12
Screenshot 2024-06-25 at 19 24 24


for claim, it says `0` so it is not yet ok

Screenshot 2024-06-25 at 19 25 26

**Comment by omykhailovskyi** [comment]
@tomasklim to discuss with @sime either we need to display ETH zero amount? Or hide them? for Replaced tx.

**Comment by tomasklim** [comment]
Closing in favor of #13084, please check it if it is ok

**Pull request closed without merging**